specifyweb/backend/businessrules/uniqueness_rules.py

```diff
@@ -136,7 +152,46 @@ def get_exception(conflicts, matchable, field_map):
         if len(matchable.keys()) == 0 or set(all_fields) != set(field_map.keys()):
             continue
 
-        conflicts = model.objects.only('id').filter(**matchable)
+        conflicts_query = model.objects.only('id')
+
+        db_alias = router.db_for_read(model)
+        connection = connections[db_alias]
+
+        filter_kwargs = dict(matchable)
+
+        apply_case_sensitive = connection.vendor == 'mysql' and not rule.isDatabaseConstraint
+
+        if apply_case_sensitive:
+            annotations: dict[str, Func] = {}
+            transformed_filters: dict[str, object] = {}
+
+            case_sensitive_field_types = (
+                dj_models.CharField,
+                dj_models.TextField,
+                dj_models.SlugField,
+                dj_models.EmailField,
+                dj_models.UUIDField,
+            )
+
+            for index, (field_path, value) in enumerate(matchable.items()):
+                if not isinstance(value, str):
+                    continue
+                resolved_field = resolve_model_field(model, field_path)
+                if isinstance(resolved_field, case_sensitive_field_types):
+                    alias = f"__binary_filter__{index}"
+                    annotations[alias] = Func(
+                        F(field_path),
+                        function='BINARY',
+                        output_field=dj_models.TextField(),
+                    )
+                    transformed_filters[alias] = value
+                    filter_kwargs.pop(field_path, None)
+
+            if annotations:
+                conflicts_query = conflicts_query.annotate(**annotations)
+                filter_kwargs.update(transformed_filters)
```

Comment:
(optional) 

Similar to https://github.com/specify/specify7/pull/7450#discussion_r2382962118: 

Can you extract this case-sensitivity logic into one or more separate functions?
I imagine it could look something like: 

```py
def validate_unique(model, instance): 
    .... # Previous validate unique code

    if apply_case_sensitive: 
        conflicts_query, applied_case_sensitive_filters = case_sensitive_filters(conflict_query, matchable)
        filter_kwargs.update(applied_case_sensitive_filters)

    .... # Any other query transformations
    conflicts = conflicts_query.filter(**filter_kwargs)

def case_sensitive_filters(queryset, matchable): 
    # Code from above here
    ...
```